backend/src/graphql/resolver/UserResolver.ts

```diff
@@ -344,11 +358,13 @@ export class UserResolver {
     logger.info(`DbUser.findOne(email=${email}) = ${userFound}`)
 
     if (userFound) {
-      logger.info('User already exists with this email=' + email)
+      // ATTENTION: this logger-message will be exactly expected during tests
+      logger.info(`User already exists with this email=${email}`)
       // TODO: this is unsecure, but the current implementation of the login server. This way it can be queried if the user with given EMail is existent.
 
       const user = new User(communityDbUser)
       user.id = sodium.randombytes_random() % (2048 * 16) // TODO: for a better faking derive id from email so that it will be always the same id when the same email comes in?
+      user.gradidoID = '11111111-2222-4333-4444-55555555'
```

Comment:
Not sure why we set a constant here, I thaught that we need something that can identify a user like on line 397

Reply:
this default constant value will only be set to satisfy the NOT NULL of the column gradidoID in the table users or in the typeorm class User. There is no identification of a user on the gradidoID at the moment. And if you take a look in the surrounding code of the UserResolver here, this code creates a faked user instance only to transfer the firstname, lastname and email for sending a email and to satisfy that a user instance has to be returned.

Reply:
This code is to avoid email spying. When a user exists to a given email, we return a faked user object, simulating a successful user creation. Therefore `gradidoID = uuidv4()` should do it here. If we use the constant, you can identify the faked user and know, that the email exists

Reply:
Yes that's right, to avoid email spying it is better to use uuidv4() instead of a constant. It is changed.